datadog/resource_datadog_synthetics_test_.go

```diff
@@ -46,7 +46,22 @@ func resourceDatadogSyntheticsTest() *schema.Resource {
 				},
 				ValidateFunc: validateEnumValue(datadogV1.NewSyntheticsTestDetailsSubTypeFromValue),
 			},
-			"request": syntheticsTestRequest(),
+			"request": {
+				Description:   "The synthetics test request. Required if `type = \"api\"` and `subtype = \"http\"`.",
+				Deprecated:    "Define `request_definition` list with one element instead.",
+				ConflictsWith: []string{"request_definition"},
+				Type:          schema.TypeMap,
+				Optional:      true,
+				Elem:          syntheticsTestRequest(),
+			},
+			"request_definition": {
+				Description:   "The synthetics test request. Required if `type = \"api\"` and `subtype = \"http\"`.",
```

Comment:
This part may be legacy but `Required if `type = \"api\"` and `subtype = \"http\"`.` is not true, it should be just `Required if `type = \"api\"``